tests/pack/indexer.c

```diff
@@ -60,6 +68,22 @@ void test_pack_indexer__out_of_order(void)
 	git_indexer_free(idx);
 }
 
+void test_pack_indexer__missing_trailer(void)
+{
+	git_indexer *idx = 0;
+	git_transfer_progress stats = { 0 };
+
+	cl_git_pass(git_indexer_new(&idx, ".", 0, NULL, NULL, NULL));
+	cl_git_pass(git_indexer_append(
+		idx, missing_trailer_pack, missing_trailer_pack_len, &stats));
```

Comment:
I'd prefer if you used another pack where we already verify that we're able to parse it correctly and just trim the pack_len by 1. Like that, we make sure that we're really failing because of the damaged trailer and not because the pack itself is broken.

Reply:
Good idea, done! Had to change it to -20, because for some reason -1 wasn't enough to trigger the leak.